sql/type.go

```diff
@@ -63,8 +63,10 @@ type Type interface {
 	// Compare returns an integer comparing two values.
 	// The result will be 0 if a==b, -1 if a < b, and +1 if a > b.
 	Compare(interface{}, interface{}) (int, error)
-	// Convert a value of a compatible type to a most accurate type.
-	Convert(interface{}) (interface{}, error)
+	// Convert a value of a compatible type to a most accurate type, returning
+	// the new value, whether the value was coerced from out of range, or an
+	// error.
+	Convert(interface{}) (interface{}, bool, error)
```

Comment:
For calls with >2 return variables like this where the 2nd variable is a generic bool, should we add names?
```
Convert(interface{}) (result interface{}, isValueCoerced bool, err error)
```

Reply:
`bool` was replaced with `ConvertInRange`, ambiguity gone, resolving.